src/Analyzer/EtSchemaAnalyzer.php

```diff
@@ -0,0 +1,367 @@
+
+    /**
+     * Register removed field
+     *
+     * @param string $moduleName
+     * @param string $recordName
+     * @param string $fieldName
+     * @return array
+     */
+    private function removedField(string $moduleName, string $recordName, string $fieldName): array
+    {
+        return [
+            'level' => self::$actions[__FUNCTION__]['level'],
+            'code' => self::$actions[__FUNCTION__]['code'],
+            'location' => sprintf(
+                'urn:magento:module:%s:etc/et_schema.xml %s:%s',
+                $moduleName,
+                $recordName,
+                $fieldName
+            ),
+            'target' => $recordName,
+            'reason' => sprintf(self::$actions[__FUNCTION__]['message'], $fieldName, $recordName)
+        ];
+    }
+
+    /**
+     * Register a new field
+     *
+     * @param string $moduleName
+     * @param string $recordName
+     * @param string $fieldName
+     * @return array
+     */
+    private function addedField(string $moduleName, string $recordName, string $fieldName): array
+    {
+        return [
+            'level' => self::$actions[__FUNCTION__]['level'],
+            'code' => self::$actions[__FUNCTION__]['code'],
+            'location' => sprintf(
+                'urn:magento:module:%s:etc/et_schema.xml %s:%s',
+                $moduleName,
+                $recordName,
+                $fieldName
+            ),
+            'target' => $recordName,
+            'reason' => sprintf(self::$actions[__FUNCTION__]['message'], $fieldName, $recordName)
+        ];
+    }
+
+    /**
+     * Register field change
+     *
+     * @param string $moduleName
+     * @param string $recordName
+     * @param string $fieldName
+     * @return array
+     */
+    private function changedField(string $moduleName, string $recordName, string $fieldName): array
+    {
+        return [
+            'level' => self::$actions[__FUNCTION__]['level'],
+            'code' => self::$actions[__FUNCTION__]['code'],
+            'location' => sprintf(
+                'urn:magento:module:%s:etc/et_schema.xml %s:%s',
+                $moduleName,
+                $recordName,
+                $fieldName
+            ),
+            'target' => $recordName,
+            'reason' => sprintf(self::$actions[__FUNCTION__]['message'], $fieldName, $recordName)
+        ];
+    }
+
+    /**
+     * Analyze record structure
+     *
+     * @param string $moduleName
+     * @param $beforeRecord
+     * @param $afterRecord
+     * @return array
+     */
+    private function analyzeRecord(string $moduleName, $beforeRecord, $afterRecord): array
+    {
+        $changes = [];
+        $commonFields = array_intersect(
+            array_keys($beforeRecord['field']),
+            array_keys($afterRecord['field'])
+        );
+        foreach ($commonFields as $fieldName) {
+            if (
+                $beforeRecord['field'][$fieldName]['type'] != $afterRecord['field'][$fieldName]['type']
```

Comment:
Can we change it to strict comparison `!==` here and at other places below?

Reply:
fixed